go/vt/vtgate/planbuilder/testdata/reference_cases.json

```diff
@@ -0,0 +1,644 @@
+        "Table": "`user`, ambiguous_ref_with_source",
+        "Variant": "Scatter"
+      },
+      "Original": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source join user",
+      "QueryType": "SELECT",
+      "TablesUsed": [
+        "main.ambiguous_ref_with_source",
+        "user.user"
+      ]
+    }
+  },
+  {
+    "comment": "ambiguous unqualified reference table can merge with other opcodes left to right and vindex value is in the plan",
+    "query": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source join (select aa from user where user.id=1) user",
+    "v3-plan": {
+      "Instructions": {
+        "Inputs": [
+          {
+            "FieldQuery": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source where 1 != 1",
+            "Keyspace": {
+              "Name": "main",
+              "Sharded": false
+            },
+            "OperatorType": "Route",
+            "Query": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source",
+            "Table": "ambiguous_ref_with_source",
+            "Variant": "Reference"
+          },
+          {
+            "FieldQuery": "select 1 from (select aa from `user` where 1 != 1) as `user` where 1 != 1",
+            "Keyspace": {
+              "Name": "user",
+              "Sharded": true
+            },
+            "OperatorType": "Route",
+            "Query": "select 1 from (select aa from `user` where `user`.id = 1) as `user`",
+            "Table": "`user`",
+            "Values": [
+              "INT64(1)"
+            ],
+            "Variant": "EqualUnique",
+            "Vindex": "user_index"
+          }
+        ],
+        "JoinColumnIndexes": "L:0",
+        "OperatorType": "Join",
+        "TableName": "ambiguous_ref_with_source_`user`",
+        "Variant": "Join"
+      },
+      "Original": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source join (select aa from user where user.id=1) user",
+      "QueryType": "SELECT"
+    },
+    "gen4-plan": {
+      "Instructions": {
+        "FieldQuery": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source, (select aa from `user` where 1 != 1) as `user` where 1 != 1",
+        "Keyspace": {
+          "Name": "user",
+          "Sharded": true
+        },
+        "OperatorType": "Route",
+        "Query": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source, (select aa from `user` where `user`.id = 1) as `user`",
+        "Table": "`user`, ambiguous_ref_with_source",
+        "Values": [
+          "INT64(1)"
+        ],
+        "Variant": "EqualUnique",
+        "Vindex": "user_index"
+      },
+      "Original": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source join (select aa from user where user.id=1) user",
+      "QueryType": "SELECT",
+      "TablesUsed": [
+        "main.ambiguous_ref_with_source",
```

Comment:
Instruction seems right but not sure this is?

Reply:
same. should not refer to `main` keyspace.

Reply:
same as above. it most definitely should say which ks that table lives on